### ISSUE TITLE
What does `falling back to local: Insufficient capacity across` mean?

### ISSUE THREAD

**Issue by sefeng211** [mozilla/sccache#708]
I am getting lots of `Could not perform distributed compile, falling back to local: Insufficient capacity across 1 available servers: Insufficient capacity across 1 available servers: Failed to allocate job` error in my home sccache distribution setup.  

I tried to read the code but I can't understand :(. 

Any help is appreciated!

**Comment by froydnj** [comment]
So the code that issues that message lives here:

https://github.com/mozilla/sccache/blob/18855f4e0fc5dcbe38a26586e965a391693a2662/src/bin/sccache-dist/main.rs#L567-L650

That code is determining the best server to distribute a new job to, based on the scheduler's known jobs that have already been distributed to the server, and the cores available on each server.  That message is being issued because we think that there's no suitable server to distribute to -- either there are no servers or all the existing servers are overloaded with jobs.  Obviously the first one doesn't apply here, so it must think there are too many jobs for the existing server.

Are you compiling with a really high `-j` number, so that the scheduler thinks there's a very long queue for the build server?  The logic in that function is not super-robust, but I don't think it should be falling over...

**Comment by sefeng211** [comment]
Yeah, I was compiling a with a larger `-j`, I think I used `27` or something when my `dist-status` showed `8` cores. I also think that some of the jobs had stuck in `in_progress`. Maybe they explains it?

**Comment by froydnj** [comment]
> Yeah, I was compiling a with a larger `-j`, I think I used `27` or something when my `dist-status` showed `8` cores. I also think that some of the jobs had stuck in `in_progress`. Maybe they explains it?

If things are stuck in `in_progress`, but not actually `in_progress`, that would be a problem, yes.  Unfortunately, I don't know of a good way to reproduce that failure to investigate. :(
